iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -131,20 +134,68 @@ public static void updateHmsTableForIcebergTable(
     tbl.setParameters(parameters);
   }
 
-  private static void setCommonParameters(
-      String newMetadataLocation,
-      String uuid,
-      Set<String> obsoleteProps,
-      String currentLocation,
-      Map<String, String> parameters,
-      String tableType,
-      Schema schema,
+  public static SortOrder getSortOrder(Properties props, Schema schema) {
+    String sortOrderJsonString = props.getProperty(TableProperties.DEFAULT_SORT_ORDER);
+    return Strings.isNullOrEmpty(sortOrderJsonString) ? SortOrder.unsorted() : SortOrderParser.fromJson(schema,
+        sortOrderJsonString);
+  }
+
+  /**
+   * Calculates the properties we would like to send to the catalog.
+   * <ul>
+   * <li>The base of the properties is the properties stored at the Hive Metastore for the given table
+   * <li>We add the {@link HiveRESTCatalogClient#LOCATION} as the table location
+   * <li>We add the {@link HiveRESTCatalogClient#NAME} as
+   * TableIdentifier defined by the database name and table name
+   * <li>We add the serdeProperties of the HMS table
+   * <li>We remove some parameters that we don't want to push down to the Iceberg table props
+   * </ul>
+   * @param hmsTable Table for which we are calculating the properties
+   * @return The properties we can provide for Iceberg functions
+   */
+  public static Properties getCatalogProperties(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
```

Comment:
is that a copy&paste of HiveIcebergMetaHook#getCatalogProperties?

Reply:
Yes, it was a copy from HiveIcebergMetaHook.
Removed from `HiveIcebergMetaHook` and made reuse of the implementation in `HMSTablePropertyHelper`.

Reply:
Done.